src/components/CommentsList.tsx

```diff
@@ -0,0 +1,165 @@
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [loadError, setLoadError] = useState(false);
+  const [addError, setAddError] = useState(false);
+  const [deleteError, setDeleteError] = useState(false);
+  const [isFormVisible, setIsFormVisible] = useState(false);
+
+  const getPostCommentsFromServer = async () => {
+    try {
+      setComments([]);
+      setIsFormVisible(false);
+      setLoadError(false);
+      setIsLoading(true);
+      const commentsFromServer = await getComments(postId);
+
+      setComments(commentsFromServer);
+    } catch {
+      setLoadError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const postNewComment = useCallback(async (comment: CommentData) => {
+    const newComment = { ...comment, postId };
+
+    try {
+      const addedComment = await postComment(newComment);
+
+      setComments(currentComment => [...currentComment, addedComment]);
+    } catch {
+      setAddError(true);
+    }
+  }, []);
+
+  const handleDeleteComment = useCallback(async (commentId: number) => {
+    const tempComments = [...comments];
+
+    setComments(currentComments => (
+      currentComments.filter(comment => comment.id !== commentId)
+    ));
+
+    try {
+      setDeleteError(false);
+      await deleteComment(commentId);
+    } catch {
+      setComments(tempComments);
+      setDeleteError(true);
+    }
+  }, []);
+
+  useEffect(() => {
+    getPostCommentsFromServer();
+  }, [postId]);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setDeleteError(false);
+      setAddError(false);
+    }, 3000);
+
+    return () => clearTimeout(timer);
+  }, [addError, deleteError]);
+
+  if (isLoading) {
+    return <Loader />;
+  }
+
+  if (loadError) {
+    return (
+      <div className="notification is-danger" data-cy="CommentsError">
+        Something went wrong
+      </div>
+    );
+  }
+
+  return (
+    <>
+      {!comments.length
+        ? (
+          <p className="title is-4" data-cy="NoCommentsMessage">
+            No comments yet
+          </p>
+        )
+        : (
+          <>
+            <p className="title is-4">Comments:</p>
+
+            {comments.map(comment => (
+              <article
+                className="message is-small"
+                data-cy="Comment"
+                key={comment.id}
+              >
+                <div className="message-header">
+                  <a
+                    href={`mailto:${comment.email}`}
+                    data-cy="CommentAuthor"
+                  >
+                    {comment.name}
+                  </a>
+                  <button
+                    data-cy="CommentDelete"
+                    type="button"
+                    className="delete is-small"
+                    aria-label="delete"
+                    onClick={() => handleDeleteComment(comment.id)}
+                  >
+                    delete button
```

Comment:
Remove the text 'delete button' inside the button element, as it is not necessary and could confuse users.